City-of-Helsinki/helsinki-paatokset
# UHF-8391: Update to Drupal 10

# [UHF-8391](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8391)


## What was done


* Upgrade to Drupal 10, fix deprecated code.

In addition, this PR contains the following fixes to other issues that I found while testing.
- Fixes calling strip_tags() with null in CaseService. This is deprecated in PHP.
  - 75b99fe683d96678e0ce70fda1d5d4a5b0981e06
- Fixes use of dynamic properties. This is deprecated in PHP.
  - d2fdc57dadd2cad3898e194021e634f72d818a77
  - 748c866c10861be3250580c37a4a8bd46c75c9eb
- Fix visual bug in menu that I found while testing. The menu looks weird on tabled sized devices.
  - b702d251807f8cc0c57fc8bbe7abf69896b0032d
- Fix crash on policymaker page when elasticsearch is down
  - ec799f9d11b8754a4f040fb4a87fe4b2a8865602
- Remove dead code
  - 3d9dc5361bdb2de190ed04936b55bb9d9436b870
- Fix crash on /admin/config/development/logging
  - ccbcc8da39a1d81780db04311bbc48920357e859 

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-8391-drupal-10`
  * `make new`
* Run `make drush-cr`
* Enable maximum logging at [/admin/config/development/logging](http://helsinki-paatokset.docker.so/fi/admin/config/development/logging).

## How to test


We should try to run as much of the custom code as possible and watch for crashes or unwanted behaviour.

* [x] Run install instructions from v3 PR (#319):

```sh
# Install test content
drush en paatokset_test_content

# Import decisionmakers
drush migrate-import ahjo_decisionmakers:all
drush migrate-import ahjo_decisionmakers:all_sv

# Import trustees
drush migrate-import ahjo_trustees:all
drush migrate-import ahjo_trustees:all_sv

# Import (latest) meetings
drush migrate-import ahjo_meetings:latest
drush migrate-import ahjo_decisions:latest

# Run TPR migrations
drush en -y migrate_tools
drush migrate-import tpr_unit --limit 100
drush migrate-import tpr_service --limit 100

# Fetch cases and decisions
drush ap:update organization U3212002040VH1 -v
drush ap:update cases hel-2014-000497 -v
drush ap:update cases hel-2016-012511 -v
drush ap:update decisions b4b8e27f-9763-4fd0-810c-96ada6c1c4b7 -v
drush ap:update decisions 2b1c3a6c-f0f2-411e-b6f0-23b56cd0e0ea -v
drush ap:update decisions afe79ec8-328f-414d-85c6-35de280ef0a9 -v
drush ap:update decisions 0e036367-174a-47fb-8ac5-37f923981c16 -v
drush ap:update decisions ffe61d01-9ecb-47b7-a5b3-4eaf1ebe598d -v
drush ap:update decisions fb53d9ca-efaa-4aa2-b417-fe09efeea58a -v
drush ap:update decisions 5940cd98-f540-40b1-947f-c76ff233b93e -v
drush ap:update decisions 20af8888-1394-4c84-96ac-43f78ff8ff4c -v

# Get motions, update decison data, fetch missing cases and decisions
drush ap:gm -v
drush ap:ud -v
drush queue:run ahjo_api_retry_queue -v
drush ap:ud --logic=case -v
drush ap:ud --logic=meeting -v
```

* [x] Browse the page, click buttons
* [x] Edit nodes
* [x] Try to edit text using the new CKeditor5
* [x] If possible, test the upgrade process with the production database

## Designers review




[UHF-8391]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8391?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
